samza-sql/src/main/java/org/apache/samza/sql/testutil/SamzaSqlQueryParser.java

```diff
@@ -0,0 +1,182 @@
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.testutil;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.calcite.config.Lex;
+import org.apache.calcite.jdbc.CalciteConnection;
+import org.apache.calcite.plan.Contexts;
+import org.apache.calcite.plan.ConventionTraitDef;
+import org.apache.calcite.plan.RelTraitDef;
+import org.apache.calcite.rel.RelCollationTraitDef;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.sql.SqlAsOperator;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlInsert;
+import org.apache.calcite.sql.SqlJoin;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlSelect;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.tools.FrameworkConfig;
+import org.apache.calcite.tools.Frameworks;
+import org.apache.calcite.tools.Planner;
+import org.apache.samza.SamzaException;
+
+
+/**
+ * Utility class that is used to parse the Samza sql query to figure out the inputs, outputs etc..
+ */
+public class SamzaSqlQueryParser {
+
+  private SamzaSqlQueryParser() {
+  }
+
+  public static class QueryInfo {
+    private final List<String> inputSources;
+    private String selectQuery;
+    private String outputSource;
+
+    public QueryInfo(String selectQuery, List<String> inputSources, String outputSource) {
+      this.selectQuery = selectQuery;
+      this.outputSource = outputSource;
+      this.inputSources = inputSources;
+    }
+
+    public List<String> getInputSources() {
+      return inputSources;
+    }
+
+    public String getSelectQuery() {
+      return selectQuery;
+    }
+
+    public String getOutputSource() {
+      return outputSource;
+    }
+  }
+
+  public static QueryInfo parseQuery(String sql) {
+
+    Pattern insertIntoSqlPattern = Pattern.compile("insert into (.*) (select .* from (.*))", Pattern.CASE_INSENSITIVE);
```

Comment:
why don't we translate the "insert into" as a "TableAlter" node in sql? this looks weird to me, and not consistent with other relNode logic.

Reply:
Long term we need to figure out a way to handle this that is consitent with other RelNode operations. We probably need help from calcite for this. I will open a Jira to track this one.